src/GdbServer.cc

```diff
@@ -1426,10 +1426,18 @@ void GdbServer::serve_replay(const ConnectionFlags& flags) {
     timeline.set_reverse_execution_barrier_event(first_run_event);
   }
 
-  activate_debugger();
+  bool first_run = true;
+
+  while (first_run || flags.keep_listening) {
```

Comment:
This can be simplified to avoid the `first_run` variable. Just change it to `do { ... } while (flags.keep_listening);`